http.go

```diff
@@ -49,6 +50,7 @@ type Request struct {
 
 	// To detect scheme changes in redirects
 	schemaUpdate bool
+	deadline     time.Time
```

Comment:
I would prefer that instead of adding this field to ʼRequestʼ you add an extra parameter to ʼdoNonNilReqRespʼ etc.